txscript/opcode.go

```diff
@@ -1850,7 +1850,17 @@ func opcodeCheckSig(op *parsedOpcode, vm *Engine) error {
 		return nil
 	}
 
-	ok := signature.Verify(hash, pubKey)
+	var ok bool
+	if vm.sigCache != nil {
+		ok = vm.sigCache.Exists(hash, signature, pubKey)
```

Comment:
This assumes existence in the signature cache means it is a valid signature and the below only adds the result to the cache if it's a valid signature.  I'm fairly certain this means that invalid signatures will have to be re-checked which partially defeats the DoS protection aspects.